### PR TITLE
doc: add documentation of vfio plugin

### PR DESCRIPTION
**Pull request by r-rojo**

### CODE REVIEW

File: plugins/vfio/readme.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# VFIO Plugin
+
+The OPAE vfio plugin, v-opae, is an OPAE plugin designed to discover and
+interface with accelerator resources on PCIe devices bound to the vfio-pci
+kernel driver. Given that the OPAE object model currently differentiates
+between management (`FPGA_DEVICE`) and accelerator (`FPGA_ACCELERATOR`)
+objects,the vfio plugin focuses on enabling developers to create user-mode
```
Comment (tswhison) [inline plugins/vfio/readme.md:7]: s/objects,the/objects, the/

File: plugins/vfio/readme.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+driversfor accelerators using the OPAE API. This means that even when a PCIe
```
Comment (tswhison) [inline plugins/vfio/readme.md:8]: s/driversfor/drivers for/

File: plugins/vfio/readme.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+device with management IP may be bound to the vfio-pci driver, the OPAE
+operationsallowed for those resources are limited mostly to those relating
```
Comment (tswhison) [inline plugins/vfio/readme.md:10]: s/operationsallowed/operations allowed/

File: plugins/vfio/readme.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+to discovery and identification.
+
+### Sample Use Case
+As mentioned before, this plugin is best used for runninng user-space
```
Comment (tswhison) [inline plugins/vfio/readme.md:14]: s/runninng/running/

File: plugins/vfio/readme.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+and one or more accelerator functions via VF (virtual function) endpoints.
+The image below depicts such an example and shows accelerator data flow
+between a software application and the accelerator functional unit (AFU).
+
+<p align="center">
+  <img width="640" height="480" src="vfio-plugin-example.svg">
+</p>
+
+#### Binding vfio-pci Driver
+Using the example above, the folowing shows how one may bind the vfio-pci
```
Comment (tswhison) [inline plugins/vfio/readme.md:26]: s/folowing/following/

File: plugins/vfio/readme.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+and the device IDs for the PF and VF are aba0 and aba1 respectively.
+
+Load the vfio-pci driver and create one VF.
+```shell
+> sudo modprobe vfio-pci
+> echo 1 | sudo tee /sys/bus/pci/devices/0000\:ab\:00.0/sriov_numvfs
+```
+
+At this point, if this is the first VF created for the device, a
+sysfs node will be created for using function 1. The new PCIe
```
Comment (tswhison) [inline plugins/vfio/readme.md:37]: s/for using/for it using/

File: plugins/vfio/readme.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+```
+
+More information about vfio can be found
+[here](https://www.kernel.org/doc/Documentation/vfio.txt).
+
+#### Enabling the Plugin
+Currently, the vfio plugin will not automatically load. One must use the OPAE
+configuration file in order to enable it. The example below shows how to
+enable it by adding a configuration item into the configurations dictionary
+(as identified by the `vfio` name) and by adding the idendtifier to the list
```
Comment (tswhison) [inline plugins/vfio/readme.md:57]: s/idendtifier/identifier/

File: python/opae.admin/opae.admin.changes [hunk 8]
```diff
@@ -0,0 +1,118 @@
+diff --git a/python/opae.admin/opae/admin/sysfs.py b/python/opae.admin/opae/admin/sysfs.py
```
Comment (tswhison) [inline python/opae.admin/opae.admin.changes:1]: Not sure you meant to submit this file.  Remove it?

**Review by tswhison — APPROVED** [review]

**Review by anandaravuri — APPROVED** [review]

**Pull request merged**